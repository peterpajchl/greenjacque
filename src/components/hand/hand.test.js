import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import sinon from 'sinon';
import { shallow } from 'enzyme';
import Hand from './hand';
import {
  assembleDeck
} from '../../common/common';
import Card from '../../common/card';

Enzyme.configure({ adapter: new Adapter() });

describe('<Hand /> component', () => {
  const deck = assembleDeck();
  const cards = deck.slice(0, 5);

  it('should match the snapshot', () => {
    const handleSort = sinon.spy();
    const wrapper = shallow(<Hand cards={cards} sort={handleSort} />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render 5 cards when 5 cards drawn', () => {
    // should also check order
    const handleSort = sinon.spy();
    const wrapper = shallow(<Hand cards={cards} sort={handleSort} />);
    expect(wrapper.find('.drawn-card').length).toEqual(5);
  });

  it('should render enabled sort button', () => {
    const handleSort = sinon.spy();
    const wrapper = shallow(<Hand cards={cards} sort={handleSort} />);
    const button = wrapper.find('button[type="button"]');
    expect(button.length).toEqual(1);
    expect(button.text()).toEqual('Sort');
    expect(button.hasClass('btn control')).toBeTruthy();
    expect(button.props().disabled).toBe(false);
  });

  it('should render sort button with disabled state for less than 2 cards', () => {
    const handleSort = sinon.spy();
    const wrapper = shallow(<Hand cards={[new Card('H', '2')]} sort={handleSort} />);
    const button = wrapper.find('button[type="button"]');
    expect(button.props().disabled).toBe(true);
  });

  it('should allow to call sort', () => {
    const handleSort = sinon.spy();
    const wrapper = shallow(<Hand cards={[new Card('H', '2')]} sort={handleSort} />);
    const button = wrapper.find('button[type="button"]');
    button.simulate('click');
    expect(handleSort).toHaveProperty('callCount', 1);
  });

});