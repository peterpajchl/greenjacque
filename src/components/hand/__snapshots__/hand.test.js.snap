// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hand /> component should match the snapshot 1`] = `"<div class=\\"main-body__block main-body__block--hand\\"><div class=\\"body-header\\"><h3 class=\\"body-header__title\\">Hand</h3><div class=\\"body-header__counter\\">5 (0)</div></div><ul class=\\"cards-in-hand__list\\"><li class=\\"drawn-card slide-in\\" style=\\"left:0px;z-index:1;transform:translateX(-100%)\\"><img class=\\"drawn-card__image\\" src=\\"./assets/cards/2C.svg\\" alt=\\"Card: 2 of C\\" width=\\"100%\\"/></li><li class=\\"drawn-card slide-in\\" style=\\"left:28px;z-index:2;transform:translateX(-200%)\\"><img class=\\"drawn-card__image\\" src=\\"./assets/cards/3C.svg\\" alt=\\"Card: 3 of C\\" width=\\"100%\\"/></li><li class=\\"drawn-card slide-in\\" style=\\"left:56px;z-index:3;transform:translateX(-300%)\\"><img class=\\"drawn-card__image\\" src=\\"./assets/cards/4C.svg\\" alt=\\"Card: 4 of C\\" width=\\"100%\\"/></li><li class=\\"drawn-card slide-in\\" style=\\"left:84px;z-index:4;transform:translateX(-400%)\\"><img class=\\"drawn-card__image\\" src=\\"./assets/cards/5C.svg\\" alt=\\"Card: 5 of C\\" width=\\"100%\\"/></li><li class=\\"drawn-card slide-in\\" style=\\"left:112px;z-index:5;transform:translateX(-500%)\\"><img class=\\"drawn-card__image\\" src=\\"./assets/cards/6C.svg\\" alt=\\"Card: 6 of C\\" width=\\"100%\\"/></li></ul><div class=\\"card-controls\\"><button class=\\"btn control\\" type=\\"button\\">Sort</button></div></div>"`;
